### PR TITLE
Refactor cjh 09052023

### PR DESCRIPTION
**Pull request by ChrisHobbs1992**
Refactored person methods. 

Haven't removed Sam's original method's from the project yet.

Person search was not in scope of these changes.

### CODE REVIEW

File: PersonSearchDetails/PageObjects/Person/PersonMethods.cs [hunk 1]
```diff
@@ -4,38 +4,35 @@
...
-            xrmBrowser.ThinkTime(1000);
-            driver.FindElement(By.XPath("//*[@id=\"address1_stateorprovince_cl\"]")).Click();
-            driver.FindElement(By.XPath("//*[@id=\"address1_stateorprovince_i\"]")).SendKeys(county);
-            driver.FindElement(By.XPath("//*[@id=\"address1_stateorprovince_i\"]")).SendKeys(Keys.Return);
-            xrmBrowser.ThinkTime(1000);
-            driver.FindElement(By.XPath("//*[@id=\"address1_postalcode_cl\"]")).Click();
-            driver.FindElement(By.XPath("//*[@id=\"address1_postalcode_i\"]")).SendKeys(postCode);
-            driver.FindElement(By.XPath("//*[@id=\"address1_postalcode_i\"]")).SendKeys(Keys.Return);
+
+            throw new NotImplementedException("");
```
Comment (samGreenwood2022) [inline PersonSearchDetails/PageObjects/Person/PersonMethods.cs:28]: Not an issue but just wondering what happened to this code, looks like its been added but cant be found anywhere in my local repo on your branch

File: PersonSearchDetails/StepDefinitions/Person/PersonCreateStepDefinitions.cs [hunk 2]
```diff
@@ -110,29 +136,63 @@ public void ThenTheExpectedAuditEventsWouldBeCreated()
...
             driver.SwitchTo().Frame("id = \"InlineDialog_Iframe\"");
             xrmBrowser.ThinkTime(1000);
+
+            //Locate the error element and pull back the text
+            //This needs a new name and some description
             String errorTextd = driver.FindElement(By.XPath("//*[@id=\"ErrorTitle\"]")).Text;
             xrmBrowser.ThinkTime(1000);
+
+            //Confirm Duplicate record is the text pulled back
+            //Query: Is this too brittle?
```
Comment (samGreenwood2022) [inline PersonSearchDetails/StepDefinitions/Person/PersonCreateStepDefinitions.cs:187]: I couldnt actually properly test this at the time, now that i know the duplication rules are in place for NHS Number i can use that to complete the test, i'll look at this as soon as i finish my scenario 8 bits

File: PersonSearchDetails/PageObjects/Person/PersonMethods.cs (hunk 1 above)
Comment (ChrisHobbs1992) [reply PersonSearchDetails/PageObjects/Person/PersonMethods.cs:28]: The exception was unfinished - I've now amended it to have some text. The overall method has been extracted into individual test step methods, so I've added the exception to show this is no longer relevant. 

We should delete the method (and possibly the class) in a future clean up.

File: PersonSearchDetails/StepDefinitions/Person/PersonAmendStepDefinitions.cs [hunk 3]
```diff
@@ -34,90 +35,102 @@ public class PersonAmendStepDefinitions
             xrmBrowser = browser;
         }
 
+
+        /////a known person already exists in the system <firstname> and <dob> and <dateMovedIn> and <ethnicity> and <gender> and <preferredLanguage>
```
[the remark this answers is not in the document]
Comment (ChrisHobbs1992) [reply PersonSearchDetails/StepDefinitions/Person/PersonAmendStepDefinitions.cs:39]: Agreed and done.

File: PersonSearchDetails/StepDefinitions/Person/PersonAmendStepDefinitions.cs [hunk 4]
```diff
@@ -34,90 +35,102 @@ public class PersonAmendStepDefinitions
...
-            xrmBrowser.ThinkTime(1000);
             // select the correct iFrame
             driver.SwitchTo().Frame("contentIFrame1");
-            driver.FindElement(By.XPath("//*[@id=\"firstname\"]/div[1]")).Click();
-            driver.FindElement(By.XPath("//*[@id=\"firstname_i\"]")).SendKeys(firstname);
-            driver.FindElement(By.XPath("//*[@id=\"lastname\"]/div[1]")).Click();
+
+
+
+            Page_PersonCoreDemographics.EnterFirstName(driver, firstname);
```
[the remark this answers is not in the document]
Comment (ChrisHobbs1992) [reply PersonSearchDetails/StepDefinitions/Person/PersonAmendStepDefinitions.cs:57]: Agreed - sorted

**Review by samGreenwood2022 — APPROVED** [review]

**Pull request merged**